VideoAPI/Video.API/Controllers/ConferenceManagementController.cs

```diff
@@ -97,5 +98,45 @@ public async Task<IActionResult> EndVideoHearingAsync(Guid conferenceId)
                 return StatusCode(ex.StatusCode, ex.Response);
             }
         }
+        
+        /// <summary>
+        /// Transfer a participant in or out of a hearing
+        /// </summary>
+        /// <param name="conferenceId">Id for conference</param>
+        /// <param name="transferRequest">Participant and direction of transfer</param>
+        /// <returns></returns>
+        [HttpPost("{conferenceId}/transfer")]
+        [SwaggerOperation(OperationId = "TransferParticipant")]
+        [ProducesResponseType((int) HttpStatusCode.OK)]
+        public async Task<IActionResult> TransferParticipantAsync(Guid conferenceId, TransferParticipantRequest transferRequest)
+        {
+            _logger.LogDebug("TransferParticipant");
+            try
+            {
+                switch (transferRequest.TransferType)
+                {
+                    case TransferType.Call:
+                        _logger.LogDebug($"Transferring participant {transferRequest.ParticipantId} into hearing room");
+                        await _videoPlatformService.TransferParticipantAsync(conferenceId,
+                            transferRequest.ParticipantId,
+                            RoomType.WaitingRoom, RoomType.HearingRoom);
+                        break;
+                    case TransferType.Dismiss:
+                        _logger.LogDebug(
+                            $"Transferring participant {transferRequest.ParticipantId} out of hearing room");
+                        await _videoPlatformService.TransferParticipantAsync(conferenceId,
+                            transferRequest.ParticipantId,
+                            RoomType.HearingRoom, RoomType.WaitingRoom);
+                        break;
+                }
+                
+                return Ok();
+            }
+            catch (KinlyApiException ex)
+            {
```

Comment:
if the endpoint is called and the user is already in the destination room (already been called) is an exception thrown? what type?

Reply:
No exception thrown. Even invalid ids can be passed. Kinly will return Accepted unless there is an obvious error. Unfortunately even invalid property names are not detected so we need to follow the swagger document closely found [here](https://test.hearings.hmcts.net/swagger-ui/index.html?configUrl=/api/docs/swagger-config#/Hearing/transfer).

NOTE: the API calls actually return 202 despite the document stating it returns OK